tls/s2n_handshake_io.c

```diff
@@ -1232,13 +1260,23 @@ static int s2n_handshake_read_io(struct s2n_connection *conn)
         }
 
         return S2N_SUCCESS;
-    } else if (record_type != TLS_HANDSHAKE) {
-        if (record_type == TLS_ALERT) {
+    } else if (record_type == TLS_ALERT) {
             POSIX_GUARD(s2n_process_alert_fragment(conn));
-        }
-
-        /* Ignore record types that we don't support */
 
+        /* We're done with the record, wipe it */
+        POSIX_GUARD_RESULT(s2n_wipe_record(conn));
+        return S2N_SUCCESS;
+    } else if (record_type != TLS_HANDSHAKE) {
+        /* TLS1.3 should reject unexpected record types.
+         *= https://tools.ietf.org/rfc/rfc8446#5
```

Comment:
The extra comments on compliance comments is unnecessary if the compliance comment has enough information already.